src/core/Class.js

```diff
@@ -8,110 +8,108 @@ import * as Util from './Util';
 
 // Thanks to John Resig and Dean Edwards for inspiration!
 
-export function Class() {}
-
-Class.extend = function (props) {
-
+export class Class {
 	// @function extend(props: Object): Function
 	// [Extends the current class](#class-inheritance) given the properties to be included.
 	// Returns a Javascript function that is a class constructor (to be called with `new`).
-	const NewClass = function (...args) {
+	static extend(props) {
+		class NewClass {
+			_initHooksCalled = false;
 
-		Util.setOptions(this);
+			constructor(...args) {
+				Util.setOptions(this);
 
-		// call the constructor
-		if (this.initialize) {
-			this.initialize.apply(this, args);
-		}
+				// call the constructor
+				if (this.initialize) {
+					this.initialize.apply(this, args);
+				}
+
+				// call all constructor hooks
+				this.callInitHooks();
+			}
 
-		// call all constructor hooks
-		this.callInitHooks();
-	};
+			callInitHooks() {
+				if (this._initHooksCalled) { return; }
 
-	const parentProto = this.prototype;
+				if (super.callInitHooks) {
+					super.callInitHooks.call(this);
+				}
 
-	const proto = Object.create(parentProto);
-	proto.constructor = NewClass;
+				this._initHooksCalled = true;
 
-	NewClass.prototype = proto;
+				const proto = NewClass.prototype;

```

Comment:
Managed to move the `callInitHooks()` method into the class definition, but I need to find some kind of way to refer to the prototype here without statically referencing `NewClass`.

Reply:
933fa76fb87782a618b866cb920079df7463c2e3 takes care of this.